buildkite/scripts/merges-cleanly.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+BRANCH=$1
+echo 'Testing for conflicts with `'"$BRANCH"'`...'
+
+# Adapted from this stackoverflow answer: https://stackoverflow.com/a/10856937
+# The git merge-tree command shows the content of a 3-way merge without
+# touching the index, which we can then search for conflict markers.
+
+git merge-tree `git merge-base origin/$BRANCH HEAD` origin/$BRANCH HEAD | grep "^<<<<<<<"
```

Comment:
We discussed emitting a useful diff on our call; should we try to do that? It's probably fine to clobber the checkout and do a `git merge origin/$BRANCH && git diff --exit-code`

Reply:
This also appears not to be working correctly. If I locally check out this branch and `git merge origin/feature/snapps-protocol`, I see conflicts, but CI seems contented.

Reply:
I don't think its fine to clobber the checkout, i think a nontrivial amount of the local state is saved between runs in the same pipeline, but I will see about getting better debug output. Certainly I need to see more information to understand why it doesn't catch conflicts yet

Reply:
fixed, when there are merge conflicts the lines begin with +<<<<< so the grep needs "^+<<<<"

Reply:
Congrats, you are now successfully blocked from merging this :rofl: